packages/app/src/app/pages/Sandbox/Editor/Header/Actions.tsx

```diff
@@ -29,6 +30,64 @@ const TooltipButton = ({ tooltip, ...props }) => {
   );
 };
 
+type ForkActionProps = {
+  isPrimaryAction: boolean;
+  handleFork: (teamId: string) => void;
+  handleSignIn: () => void;
+};
+
+const ForkAction = ({
+  isPrimaryAction,
+  handleFork,
+  handleSignIn,
+}: ForkActionProps) => {
+  const {
+    user,
+    editor: { isForkingSandbox, currentSandbox },
+  } = useAppState();
+  const { isPro } = useWorkspaceSubscription();
+
+  const { privacy, permissions } = currentSandbox;
+  const isUnlistedOrPrivate = privacy === 1 || privacy === 2;
+  const preventSandboxLeaving = permissions?.preventSandboxLeaving;
+
+  const variant = isPrimaryAction ? 'primary' : 'secondary';
+
+  if (user) {
+    if (!isPro && isUnlistedOrPrivate) {
+      return (
+        <TooltipButton
+          tooltip="You do not have permission to fork this sandbox"
+          variant={variant}
+          disabled
+        >
+          <ForkIcon css={css({ height: 3, marginRight: 1 })} /> Fork
+        </TooltipButton>
+      );
+    }
+
+    return (
+      <ForkButton user={user} forkClicked={handleFork} variant={variant} />
+    );
+  }
+
+  return (
+    <TooltipButton
+      tooltip={
+        preventSandboxLeaving
+          ? 'You do not have permission to fork this sandbox'
```

Comment:
Why is this particular tooltip text dependent on the `preventSandboxLeaving` rather then on user permission? Should it say something else?

Reply:
I think when it's unlisted and accessed (via link) by an anonymous user it should disable forking. When it's public and accessed by an anonymous user it handles sign in rather than forking, but doesn't show the "no permission" tooltip.